### PR TITLE
Less undefined tokens

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation fixes

Define many undefined tokens, and other misc fixes.
    
Progress towards #9411, extracted from #10118, rebased ontop of #10192.

### CODE REVIEW

**Comment by Zimmi48** [comment]
Look only at the last commit.
This is not absolutely required but I wouldn't mind getting an ack from @mattam82 (generalized rewriting chapter) and @SkySkimmer (universe constraints).

**Review by SkySkimmer — APPROVED** [review]
looks OK

File: doc/sphinx/addendum/generalized-rewriting.rst
Comment (cpitclaudel) [inline doc/sphinx/addendum/generalized-rewriting.rst]: :heart_eyes:

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (cpitclaudel) [inline doc/sphinx/addendum/universe-polymorphism.rst]: Is this preferable to `{? Polymorphic} Universe @ident`?

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (cpitclaudel) [inline doc/sphinx/user-extensions/syntax-extensions.rst]: Do you need the space here?

**Review by cpitclaudel — APPROVED** [review]
:clap:

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (Zimmi48) [reply doc/sphinx/addendum/universe-polymorphism.rst]: I think my concern was about people finding the command in the index.

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (Zimmi48) [reply doc/sphinx/user-extensions/syntax-extensions.rst]: I can try without.

**Comment by Zimmi48** [comment]
Pushed an additional commit on top.

**Comment by cpitclaudel** [comment]
The new commit looks good to me as well.  Should I merge this now, or should I wait for the PR this is based on to be merged?

**Comment by Zimmi48** [comment]
I've rebased to remove the dependent commits, although maybe this is just a waste of CI resources.

File: doc/sphinx/proof-engine/tactics.rst
Comment (JasonGross) [inline doc/sphinx/proof-engine/tactics.rst]: "refers to" seems wrong.  Do you mean "denotes an application whose head symbol is"?

File: doc/sphinx/proof-engine/tactics.rst
Comment (JasonGross) [inline doc/sphinx/proof-engine/tactics.rst]: Interesting.  Is `@scope` not a thing?

File: doc/sphinx/proof-engine/tactics.rst
Comment (JasonGross) [inline doc/sphinx/proof-engine/tactics.rst]: Silly question: what happens if I do `unfold "+"%nat%type.`?

File: doc/sphinx/proof-engine/vernacular-commands.rst
Comment (JasonGross) [inline doc/sphinx/proof-engine/vernacular-commands.rst]: Grammar: should be "presents the user *from* also *performing* a"

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (JasonGross) [inline doc/sphinx/user-extensions/syntax-extensions.rst]: Are you intending to escape the `s` here

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (JasonGross) [inline doc/sphinx/user-extensions/syntax-extensions.rst]: Do production lists not use the `{? ...}` syntax for optional things?

File: doc/sphinx/proof-engine/ssreflect-proof-language.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/ssreflect-proof-language.rst]: This is not really escaping. This is how we do to add an `s` that is outside of the token but without space between the token and the `s`.

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: Indeed, you are right:

```coq
Opaque Nat.add.
Goal 1 + 0 = 1.
unfold Nat.add.
```

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: It is but it does not denote the key but the scope name itself.

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: That's a syntax error.

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (Zimmi48) [reply doc/sphinx/user-extensions/syntax-extensions.rst]: Same answer as above.

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (Zimmi48) [reply doc/sphinx/user-extensions/syntax-extensions.rst]: They don't indeed. Maybe we'll change this in the future, but `productionlist` is a built-in Sphinx directive, whereas everything regarding the `{? ... }` syntax is coqrst-specific. BTW, maybe this should be upstreamed. WDYT @cpitclaudel?

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: That would be a lot of "denotes" :smile: Note that I didn't write any of this, so it is a bit strong to say what "I mean".

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: You are right:

```
Goal forall x, 1 * x = x.
intros.
unfold "*", "+". (* unfolds + *)
Undo.
unfold "+", "*". (* does not unfold + *)
```

I can remove the work "simultaneous" for now, but we should open an issue to clarify what the semantics is supposed to be (and maybe add some test files if they are not there already).

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: Space separated: https://coq.github.io/doc/master/refman/proof-engine/tactics.html#occurrence-sets-and-occurrence-clauses. For `unfold` at least, the previous doc (which mentioned commas) was incorrect.

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: Applied your suggestion nonetheless.

File: doc/sphinx/proof-engine/tactics.rst
Comment (JasonGross) [reply doc/sphinx/proof-engine/tactics.rst]: I think `rewrite ... at ...` uses commas?

File: doc/sphinx/user-extensions/syntax-extensions.rst
Comment (cpitclaudel) [reply doc/sphinx/user-extensions/syntax-extensions.rst]: I'm not sure.  I think they'd recommend putting it in a sphinxcontrib, and then it would increae the burden for us over putting it in coqdomain, with little to gain (on the other hand, if another project wants to use it, we could package it separately for them.  But I think we should wait until someone asks).

File: doc/sphinx/proof-engine/tactics.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: No, it uses the same syntax (space-separated numbers).

**Comment by Zimmi48** [comment]
CI failure is unrelated (overlay sync issue) and this PR is ready to go AFAICT.

**Pull request merged**
